tidb panic when run tipocket testcase in test-store

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

run tipocket testcase in test-store
bank2 -run-time=6h -tiflash-replicas=2 -tiflash-data-replicas=1 -nemesis=short_kill_tiflash_1node,shuffle-leader-scheduler,shuffle-region-scheduler,random-merge-scheduler

### 2. What did you expect to see? (Required)

success

### 3. What did you see instead (Required)

tidb panic，log：
[2021/09/03 07:08:18.763 +00:00] [ERROR] [service.go:48] ["search log panic, runtime error: slice bounds out of range [34:31], stack is goroutine 50528758 [running]:\ngithub.com/pingcap/sysutil.(*DiagnosticsServer).SearchLog.func1(0xc004f5bc48)\n\t/nfs/cache/mod/github.com/pingcap/sysutil@v0.0.0-20210730114356-fcd8a63f68c5/service.go:45 +0xa5\npanic(0x3950140, 0xc000893668)\n\t/usr/local/go/src/runtime/panic.go:965 +0x1b9\ngithub.com/pingcap/sysutil.parseLogItem(0xc0056047e0, 0x26, 0xc000610f88, 0x36ce000, 0xc006cea040)\n\t/nfs/cache/mod/github.com/pingcap/sysutil@v0.0.0-20210730114356-fcd8a63f68c5/search_log.go:308 +0x695\ngithub.com/pingcap/sysutil.readLastValidLog(0x4088390, 0xc0157bc9f0, 0xc000610f88, 0xa, 0xc007df2f00, 0x0, 0x0)\n\t/nfs/cache/mod/github.com/pingcap/sysutil@v0.0.0-20210730114356-fcd8a63f68c5/search_log.go:178 +0x8f\ngithub.com/pingcap/sysutil.resolveFiles.func1(0xc0150ae120, 0x19, 0x40a22a8, 0xc012b8e750, 0x19, 0x0)\n\t/nfs/cache/mod/github.com/pingcap/sysutil@v0.0.0-20210730114356-fcd8a63f68c5/search_log.go:89 +0x457\ngithub.com/pingcap/sysutil.resolveFiles(0x4088390, 0xc0157bc9f0, 0x7ffd6dbcdedb, 0x19, 0x17baa7bb070, 0x17baa7c9ad0, 0x0, 0x0, 0x0, 0x0, ...)\n\t/nfs/cache/mod/github.com/pingcap/sysutil@v0.0.0-20210730114356-fcd8a63f68c5/search_log.go:113 +0x335\ngithub.com/pingcap/sysutil.(*DiagnosticsServer).SearchLog(0xc00162d8d0, 0xc0040a4d90, 0x40a4578, 0xc00e583be0, 0x0, 0x0)\n\t/nfs/cache/mod/github.com/pingcap/sysutil@v0.0.0-20210730114356-fcd8a63f68c5/service.go:59 +0x126\ngithub.com/pingcap/kvproto/pkg/diagnosticspb._Diagnostics_SearchLog_Handler(0x3b30660, 0xc001c1c5a0, 0x40a1c78, 0xc0151bd800, 0x5e2ff48, 0xc014d6a000)\n\t/nfs/cache/mod/github.com/pingcap/kvproto@v0.0.0-20210806074406-317f69fb54b4/pkg/diagnosticspb/diagnosticspb.pb.go:633 +0x113\ngoogle.golang.org/grpc.(*Server).processStreamingRPC(0xc0018f8680, 0x40b0d38, 0xc00e079c80, 0xc014d6a000, 0xc001c1c630, 0x5722d00, 0x0, 0x0, 0x0)\n\t/nfs/cache/mod/google.golang.org/grpc@v1.29.1/server.go:1329 +0xcd8\ngoogle.golang.org/grpc.(*Server).handleStream(0xc0018f8680, 0x40b0d38, 0xc00e079c80, 0xc014d6a000, 0x0)\n\t/nfs/cache/mod/google.golang.org/grpc@v1.29.1/server.go:1409 +0xc68\ngoogle.golang.org/grpc.(*Server).serveStreams.func1.1(0xc00f6b2170, 0xc0018f8680, 0x40b0d38, 0xc00e079c80, 0xc014d6a000)\n\t/nfs/cache/mod/google.golang.org/grpc@v1.29.1/server.go:746 +0xab\ncreated by google.golang.org/grpc.(*Server).serveStreams.func1\n\t/nfs/cache/mod/google.golang.org/grpc@v1.29.1/server.go:744 +0xa5\n"]
[2021/09/03 07:08:18.880 +00:00] [INFO] [coprocessor.go:809] ["[TIME_COP_PROCESS] resp_time:389.460725ms txnStartTS:427465867537416194 region_id:257 store_addr:tipocket-bank2-tiflash1-tikv-0.tipocket-bank2-tiflash1-tikv-peer.tipocket-bank2-tiflash1-1630641600.svc:20160 kv_process_ms:297 kv_wait_ms:7 kv_read_ms:5 processed_versions:3687 total_versions:7516 rocksdb_delete_skipped_count:20 rocksdb_key_skipped_count:7554 rocksdb_cache_hit_count:20 rocksdb_read_count:0 rocksdb_read_byte:0"] [conn=451]



### 4. What is your TiDB version? (Required)

["Release Version"=v5.3.0-alpha-7-g4c49e5204] [Edition=Community] ["Git Commit Hash"=4c49e5204fb60b01a76b78619eb6627a1a387a71] ["Git Branch"=master] ["UTC Build Time"="2021-09-02 12:53:36"] [GoVersion=go1.16.4] ["Race Enabled"=false] ["Check Table Before Drop"=false] ["TiKV Min Version"=v3.0.0-60965b006877ca7234adaced7890d7b029ed1306]

--- a/go.mod
+++ b/go.mod
@@ -50,14 +50,14 @@ require (
 	github.com/pingcap/fn v0.0.0-20200306044125-d5540d389059
 	github.com/pingcap/kvproto v0.0.0-20211122024046-03abd340988f
 	github.com/pingcap/log v0.0.0-20210906054005-afc726e70354
-	github.com/pingcap/sysutil v0.0.0-20210730114356-fcd8a63f68c5
+	github.com/pingcap/sysutil v0.0.0-20211208032423-041a72e5860d
 	github.com/pingcap/tidb-tools v5.2.2-0.20211019062242-37a8bef2fa17+incompatible
 	github.com/pingcap/tidb/parser v0.0.0-20211011031125-9b13dc409c5e
 	github.com/pingcap/tipb v0.0.0-20211201080053-bd104bb270ba
 	github.com/prometheus/client_golang v1.5.1
 	github.com/prometheus/client_model v0.2.0
 	github.com/prometheus/common v0.9.1
-	github.com/shirou/gopsutil v3.21.2+incompatible
+	github.com/shirou/gopsutil v3.21.3+incompatible
 	github.com/shurcooL/httpgzip v0.0.0-20190720172056-320755c1c1b0
 	github.com/sirupsen/logrus v1.8.1 // indirect
 	github.com/soheilhy/cmux v0.1.4

